pkg/vm/vm.go

```diff
@@ -569,14 +569,26 @@ func (v *VM) execute(ctx *Context, op Instruction) {
 
 	// Object operations.
 	case NEWARRAY:
-		n := v.estack.Pop().BigInt().Int64()
-		items := make([]StackItem, n)
-		v.estack.PushVal(&ArrayItem{items})
+		item := v.estack.Pop()
+		switch t := item.value.(type) {
+		case *BigIntegerItem:
+			n := t.value.Int64()
+			items := make([]StackItem, n)
+			v.estack.PushVal(&ArrayItem{items})
+		case *StructItem:
+			v.estack.PushVal(&ArrayItem{t.value})
+		}
```

Comment:
I think there should be some `default:` with `panic()`. Same for `NEWSTRUCT`.

Reply:
And some tests for this behavior, of course.

Reply:
done